src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,83 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Tag.module.css';
+import Text from '../Text';
+
+export type Color =
+  | 'neutral'
+  | 'alert'
+  | 'success'
+  | 'warning'
+  | 'revise'
+  | 'brand';
+
+export const stylesByColor = {
+  neutral: styles[`tag--color-neutral`],
+  warning: styles[`tag--color-warning`],
+  brand: styles[`tag--color-brand`],
+  alert: styles[`tag--color-alert`],
+  revise: styles[`tag--color-revise`],
+  success: styles[`tag--color-success`],
+};
+
+type Props = {
+  /**
+   * The color of the tag. New colors should be supported in the flow type,
+   * in the stylesByColor object, and by its separate style in CSS file.
+   */
+  color: Color;
+  /**
+   * The child node(s) contents of the tag, nested inside the component, in addition to the icon.
+   */
+  children: React.ReactNode;
+  /**
+   * CSS class names that can be appended to the component for styling.
+   */
+  className?: string;
+  /**
+   * The tag icon (shouldn't provide color or size since those are determined
+   * by the color prop).
+   */
+  icon?: React.ReactNode;
+  /**
+   * Style of the tag.
+   */
+  variant?: 'flat' | 'outline';
```

Comment:
Kept variant as `flat` | `outline` but since `text` variant is color, should we change `color` to `variant` here and this to something else?

Reply:
Ideas:
 - Change the prop name to something else so `color` can have `variant`. We could switch to something like `hasOutline` and make it a bool, for example.
 - Talk to design about possibly deprecating the flat one. We agreed to deprecate the flat `Banner` because the "elevated" one looks good everywhere and now we'll have consistency, and I feel like `Tag` should go in the same direction. I think it's actually more important because someone might think and outlined tag means or does something differently than a flat one (like one is button and the other isn't).

Reply:
Quick update: messaged Ashley about deprecating `flat` variant, who will connect with the other designers regarding the deprecation. Waiting on updates from design

Reply:
Awesome, thanks! I'm wondering if, even if they decide to deprecate it, we'll still need to keep that variant around and do the whole deprecation procedure that the `Tooltip`'s dark variant is going through.

Reply:
Mmmm true. In that case, shouldn't be a blocker for this PR and can deprecate later